ts/screens/wallet/WalletHomeScreen.tsx

```diff
@@ -116,8 +116,18 @@ class WalletHomeScreen extends React.Component<Props, never> {
     // WIP WIP create pivotal story
     this.props.loadWallets();
     this.props.loadTransactions();
+    BackHandler.addEventListener("hardwareBackPress", this.onBackPress);
   }
 
+  public componentWillUnmount(): void {
```

Comment:
I am quite sure this can introduce a bug, try to your self: when app is started, navigate in _wallet_, then in _service_, then in _profile_. Press back (hardware) and you will notice the app goes in _messages_. The right behavior is the same in path you done, but in opposite direction: _profile (back) services (back) wallet (back) messages_

This is because when wallet is mounted, every back hardware is intercepted and handled by also this component (also RootContainer does)
So my suggestion is to place the fix in the _ScanQrCodeScreen_ and check if the cancel button does the right action